packages/idyll-document/src/index.js

```diff
@@ -187,19 +212,21 @@ class IdyllDocument extends React.PureComponent {
     const transformedSchema = mapTree(
       schema,
       node => {
+        if (typeof node === 'string') return node;
+
         // transform refs from strings to functions and store them
-        if (typeof node !== 'string' && node.ref) {
-          const refName = node.ref;
+        if (node.ref || node.hasHook) {
+          node.refName = node.ref || node.component + Math.round(Math.random() * 99999).toString();
```

Comment:
Can we change this to something deterministic? Maybe just an incrementing counter

Reply:
Sure